internal/service/domain/tree/tree_test.go

```diff
@@ -0,0 +1,831 @@
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, clusterRepo, locator)
+
+		expectedError := errors.New("GetAll failed")
+
+		treeRepo.EXPECT().GetAll(ctx).Return(nil, expectedError)
+
+		// when
+		trees, err := svc.GetAll(ctx)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, trees)
+		assert.EqualError(t, err, "500: "+expectedError.Error())
+	})
+}
+
+func TestTreeService_GetByID(t *testing.T) {
+	ctx := context.Background()
+
+	// Mocks
+	treeRepo := storageMock.NewMockTreeRepository(t)
+	sensorRepo := storageMock.NewMockSensorRepository(t)
+	imageRepo := storageMock.NewMockImageRepository(t)
+	clusterRepo := storageMock.NewMockTreeClusterRepository(t)
+	regionRepo := storageMock.NewMockRegionRepository(t)
+
+	locator := treecluster.NewLocationUpdate(clusterRepo, treeRepo, regionRepo)
+	svc := NewTreeService(treeRepo, sensorRepo, imageRepo, clusterRepo, locator)
+
+	t.Run("should return tree when found", func(t *testing.T) {
+		id := int32(1)
+		expectedTree := getTestTrees()[0]
+		treeRepo.EXPECT().GetByID(ctx, id).Return(expectedTree, nil)
+
+		// when
+		tree, err := svc.GetByID(ctx, id)
+
+		// then
+		assert.NoError(t, err)
+		assert.Equal(t, expectedTree, tree)
+	})
+
+	t.Run("should return error if tree not found", func(t *testing.T) {
+		id := int32(2)
+		expectedError := storage.ErrEntityNotFound
+		treeRepo.EXPECT().GetByID(ctx, id).Return(nil, expectedError)
+
+		// when
+		tree, err := svc.GetByID(ctx, id)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+		assert.EqualError(t, err, "404: "+expectedError.Error())
+	})
+
+	t.Run("should return error for unexpected repository error", func(t *testing.T) {
+		id := int32(3)
+		expectedError := errors.New("unexpected error")
+
+		// Set expectation for GetByID
+		treeRepo.EXPECT().GetByID(ctx, id).Return(nil, expectedError)
+
+		// when
+		tree, err := svc.GetByID(ctx, id)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+		assert.EqualError(t, err, "500: "+expectedError.Error())
+	})
+}
+
+func TestTreeService_Delete(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("should successfully delete a tree", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedTree := getTestTrees()[0]
+		expectedTree.TreeCluster = getTestTreeClusters()[0]
+
+		// Mock expectations
+		treeRepo.EXPECT().GetByID(ctx, expectedTree.ID).Return(expectedTree, nil)
+		treeRepo.EXPECT().Delete(ctx, expectedTree.ID).Return(nil)
+		locator.EXPECT().UpdateCluster(ctx, &expectedTree.TreeCluster.ID).Return(nil)
+
+		// When
+		err := svc.Delete(ctx, expectedTree.ID)
+
+		// Then
```

Comment:
Please use `// then`. You sometimes write it in capital letters and sometimes not. Let's unfiy it 👍